adapters/akka-http/src/main/scala/caliban/AkkaHttpAdapter.scala

```diff
@@ -121,16 +126,39 @@ trait AkkaHttpAdapter {
 
     get {
       extractUpgradeToWebSocket { upgrade =>
-        val (queue, source) = Source.queue[Message](0, OverflowStrategy.fail).preMaterialize()
-        val subscriptions   = runtime.unsafeRun(Ref.make(Map.empty[String, Fiber[Throwable, Unit]]))
+        val (queue: SourceQueueWithComplete[Message], source) =
+          Source.queue[Message](0, OverflowStrategy.fail).preMaterialize()
+        val subscriptions = runtime.unsafeRun(Ref.make(Map.empty[String, Fiber[Throwable, Unit]]))
         val sink = Sink.foreach[Message] {
           case TextMessage.Strict(text) =>
             val io = for {
               msg     <- Task.fromEither(json.parseWSMessage(text))
               msgType = msg.messageType
               _ <- IO.whenCase(msgType) {
                     case "connection_init" =>
-                      IO.fromFuture(_ => queue.offer(TextMessage("""{"type":"connection_ack"}""")))
+                      val ack: ZIO[Clock with Random, Throwable, Any] =
+                        IO.fromFuture(_ => queue.offer(TextMessage("""{"type":"connection_ack"}""")))
+
+                      val withKeepAlive = keepAliveTime.fold(ack) { time =>
+                        ack
+                          .flatMap(_ =>
+                            IO.fromFuture(_ =>
+                                queue
+                                  .offer(
+                                    TextMessage("""{"type":"ka"}""")
```

Comment:
The type should be `connection_keep_alive` I think?

Reply:
That would make more sense to me too, but all the protocol documents I could find said 'ka', e.g. https://medium.com/@rob.blackbourn/writing-a-graphql-websocket-subscriber-in-javascript-4451abb9cd60, https://github.com/apollographql/subscriptions-transport-ws/blob/faa219cff7b6f9873cae59b490da46684d7bea19/src/message-types.ts

Reply:
Maybe because you can potentially get tons of these and `ka` conserves bandwidth?

Reply:
You're right, they say `The keep alive message type does not follow the standard due to connection optimizations`